x/oracle/keeper/aggregator/aggregator.go

```diff
@@ -48,6 +48,34 @@ type aggregator struct {
 	dsPrices map[uint64]string
 }
 
+func (agg *aggregator) copy4CheckTx() *aggregator {
+	ret := &aggregator{
+		finalPrice:  big.NewInt(0).Set(agg.finalPrice),
+		reportPower: big.NewInt(0).Set(agg.reportPower),
+		totalPower:  big.NewInt(0).Set(agg.totalPower),
+
+		reports:  make([]*reportPrice, 0, len(agg.reports)),
+		dsPrices: agg.dsPrices,
+	}
+	for k, v := range agg.dsPrices {
+		ret.dsPrices[k] = v
+	}
+	for _, report := range agg.reports {
+		rTmp := *report
+		rTmp.price = big.NewInt(0).Set(report.price)
+		rTmp.power = big.NewInt(0).Set(report.power)
+
+		for k, v := range report.prices {
+			// prices are just record, will not be modified during execution
```

Comment:
This is not super clear. Please explain more.

Reply:
udpated